lms/djangoapps/teams/csv.py

```diff
@@ -19,3 +28,222 @@ def load_team_membership_csv(course, response):
         "Team membership CSV download is not yet implemented."
     )
     response.write(not_implemented_message + "\n")
+
+
+class TeamMemberShipImportManager(object):
+    """
+    A manager class that is responsible the import process of csv file including validation and creation of
+    team_courseteam and teams_courseteammembership objects.
+    """
+
+    def __init__(self):
+        # the list of validation errors
+        self.error_list = []
+        self.teamset_names_list = []
+        # this is a dictionary of dictionaries that ensures that a student can belong to
+        # one and only one team in a teamset
+        self.teamset_membership_dictionary = {}
+        # dictionary that matches column index to a teamset name. Used when creating teams to get the right teamset
+        self.teamset_index_dictionary = {}
+        # the currently selected user
+        self.user = ''
+        self.number_of_record_added = 0
+        # stores the course module that will be used to get course metadata
+        self.course_module = ''
+        # stores the course for which we are populating teams
+        self.course = ''
+        self.max_erros = 0
+        self.max_erros_found = False
+
+    @property
+    def import_succeeded(self):
+        """
+        Helper wrapper that tells us the status of the import
+        """
+        return len(self.error_list) == 0
+
+    def set_team_membership_from_csv(self, course, input_file):
+        """
+        Assigns team membership based on the content of an uploaded CSV file.
+        Returns true if there were no issues.
+
+        Arguments:
+            course (CourseDescriptor): Course module for which team membership needs to be set.
+        """
+        self.error_list = []
+        self.teamset_names_list = []
+        self.teamset_membership_dictionary = {}
+        self.course = course
+        self.course_module = modulestore().get_course(course.id)
+        all_rows = [row for row in csv.reader(input_file.read().decode('utf-8').splitlines())]
```

Comment:
I think a lot of this would be easier if you used a `DictReader` here: https://docs.python.org/3/library/csv.html#csv.DictReader

It would make managing the header information easier, too, e.g.:
```python
reader = DictReader(your_file)
header = reader.next()
header.pop('user')
header.pop('mode')
# now the only things left in header are the teamset names
teamset_names = {k: v for k, v in header.items()}
```

Reply:
The examples I am [finding](https://stackoverflow.com/questions/24704630/how-to-upload-and-read-csv-file-in-django-using-csv-dictreader/24704808) have a different format and I am not sure it would add any value to change it from existing implementation.

Reply:
It would make it much more readable just about everywhere.  Couple of examples:
1) `if row_data[0]:  # avoid processing rows with empty user names (excel copy and paste)` can become `if row_data['user']:` (which eliminates the need for the comment)
2) `for i in range(2, len(header_row)):` can become `for teamset_name in teamset_names:`
3) 
```python
for i in range(2, len(user_row)):
    user = user_row[0]
    team_name = user_row[i]
```
can become
```python
for teamset_name in teamset_names:
    username = user_row['user']
    team_name = user_row[teamset_name]
```